lib/simp/beaker_helpers.rb

```diff
@@ -114,9 +116,15 @@ def copy_to(sut, src, dest, opts={})
 
   # use the `puppet fact` face to look up facts on an SUT
   def pfact_on(sut, fact_name)
-    facts_json = on(sut,'puppet facts find xxx').stdout
-    facts      = JSON.parse(facts_json).fetch( 'values' )
-    facts.fetch(fact_name)
+    require 'ostruct'
+
+    if sut.which('puppet').empty?
+      fact_on(sut, fact_name, :silent => true)
+    else
+      facts_json = on(sut,'puppet facts find garbage_xxx', :silent => true).stdout
+      facts      = JSON.parse(facts_json, object_class: OpenStruct).values
+      facts.dig(*(fact_name.split('.'))) || ''
```

Comment:
There are some facts that have dots in their names:  For example (from my dev workstation):
`"cmdline": {
...
      "rd.lvm.lv": "VolGroup00/SwapVol",
...     
      "rd.shell": "0",
...
    },`

Do we care about them?

Reply:
Dot facts aren't supported in sublookups in facter so you'd have to break down the Hash independently anyway.